test/e2e/test/elasticsearch/checks_es.go

```diff
@@ -211,3 +236,84 @@ func compareMemoryLimit(topologyElement esv1.NodeSet, cgroupMemoryLimitsInBytes
+	CFSPeriodMicros int `json:"cfs_period_micros"`
+	CFSQuotaMicros  int `json:"cfs_quota_micros"`
+}) bool {
+	var expectedCPULimit *resource.Quantity
+	for _, c := range topologyElement.PodTemplate.Spec.Containers {
+		if c.Name == esv1.ElasticsearchContainerName {
+			expectedCPULimit = c.Resources.Limits.Cpu()
+		}
+	}
+	if expectedCPULimit == nil {
+		// no expected cpu, consider it's ok
+		return true
+	}
+
+	cgroupCPULimit := float64(cgroupCPU.CFSQuotaMicros) / float64(cgroupCPU.CFSPeriodMicros)
+	fmt.Println("{\"compareCPULimit\": \"", expectedCPULimit.AsApproximateFloat64() == cgroupCPULimit, "\"}")
+	fmt.Println("{\"compareCPULimit\": \"", expectedCPULimit.AsApproximateFloat64(), "\"}")
+	fmt.Println("{\"compareCPULimit\": \"", cgroupCPULimit, "\"}")
+	return expectedCPULimit.AsApproximateFloat64() == cgroupCPULimit
+}
+
+// compareResources compares the resources specified in the Elasticsearch resource with the resources
+// specified in the pods
+func compareResources(topologyElement esv1.NodeSet, pods []corev1.Pod) bool {
+	var expectedResources *corev1.ResourceRequirements
+	for _, c := range topologyElement.PodTemplate.Spec.Containers {
+		container := c
+		if c.Name == esv1.ElasticsearchContainerName {
+			expectedResources = &container.Resources
+		}
+	}
+	if expectedResources == nil {
+		expectedResources = &nodespec.DefaultResources
+	}
+	for _, pod := range pods {
+		for _, c := range pod.Spec.Containers {
+			if c.Name == esv1.ElasticsearchContainerName {
+				fmt.Println("{\"compareResources\": \"", reflect.DeepEqual(*expectedResources, c.Resources), "\"}")
+				fmt.Println("{\"compareResources\": \"", *expectedResources, "\"}")
+				fmt.Println("{\"compareResources\": \"", c.Resources, "\"}")
+				if !reflect.DeepEqual(*expectedResources, c.Resources) {
+					return false
+				}
+			}
+		}
+	}
+	return true
+}
+
+// compareStorage compares the requested storage specified in a nodeSet with the capacity of the volumes corresponding
+// to the pods of the nodeSet
+func compareStorage(topologyElement esv1.NodeSet, pvs []corev1.PersistentVolume) bool {
+	var expectedStorage *resource.Quantity
+	for _, v := range topologyElement.VolumeClaimTemplates {
+		if v.Name == volume.ElasticsearchDataVolumeName {
+			expectedStorage = v.Spec.Resources.Requests.Storage()
+		}
+	}
+	if expectedStorage == nil {
+		// no expected storage, consider it's ok
+		return true
+	}
+	for _, pv := range pvs {
+		actualStorage := pv.Spec.Capacity.Storage()
+		if pv.Spec.Local != nil {
+			// static local volume, skip
+			fmt.Println("{\"compareStorage\": \"static local volume, skip\"}")
+			continue
+		}
+		fmt.Println("{\"compareStorage\": \"", reflect.DeepEqual(expectedStorage, actualStorage), "\"}")
+		fmt.Println("{\"compareStorage\": \"", actualStorage, "\"}")
+		fmt.Println("{\"compareStorage\": \"", expectedStorage, "\"}")
+		if !reflect.DeepEqual(expectedStorage, actualStorage) {
+			return false
+		}
+	}
+	return true
+}
```

Comment:
The actual capacity of a volume might be greater than the requested capacity. It means we can't really compare a PVC and its PV. What we could do instead is comparing the claim in the `nodeSet` with the capacity claimed in the `PVC`:

```suggestion
// compareClaimedStorage compares the requested storage specified in a nodeSet with the actual capacity claimed in the PVC.
func compareClaimedStorage(topologyElement esv1.NodeSet, pvcs []corev1.PersistentVolumeClaim) bool {
	var expectedStorage *resource.Quantity
	for _, v := range topologyElement.VolumeClaimTemplates {
		if v.Name == volume.ElasticsearchDataVolumeName {
			expectedStorage = v.Spec.Resources.Requests.Storage()
			continue
		}
	}
	if expectedStorage == nil {
		// no expected storage, consider it's ok
		return true
	}
	for _, pvc := range pvcs {
		actualStorage := pvc.Spec.Resources.Requests.Storage()
		if !equality.Semantic.DeepEqual(expectedStorage, actualStorage) {
			return false
		}
	}
	return true
}
```